### ISSUE TITLE
Store auth in sqlite db.

### ISSUE THREAD

**Issue by Megaf** [minetest/minetest#6784]
Hi all. I'd like to request that auth details are stored in an indexed database instead of creating a gigantic auth.txt file.

That would prevent things like https://github.com/minetest/minetest/issues/6783 and https://github.com/minetest/minetest/issues/5334

The drawback in my opinion in doing that would be it would make maintenance more difficult. What you think?

**Comment by sorcerykid** [comment]
The auth.txt file for my server is ~180 MB. However, since rewriting the builtin authentication handler this past February with more robust storage and retrieval methods, I haven't had any performance problems whatsoever in over 10 months. In fact, max_lag rarely exceeds 0.12s even with 40+ players online.

**Comment by Megaf** [comment]
@sorcerykid By all means, do improve and make a PR so we can all try it and discuss about it.

**Comment by sfan5** [comment]
There's a mod implementing this: https://github.com/shivajiva101/sauth

The question here is whether the Minetest engine should provide this as a built-in.

**Comment by dsohler** [comment]
> The question here is whether the Minetest engine should provide this as a built-in.

Not if the mod has to be made a trusted mod and depends on 3rd party libraries that have to be installed manually by the user. Auth using a proper database instead of a key value store text file: yes.

**Comment by MarkuBu** [comment]
Maybe it is possible to use now mod storage. Then there is no need for the extra requirements

**Comment by paramat** [comment]
High priority, this really needs sorting out.

**Comment by neoascetic** [comment]
Please do something with that. It would be nice to have that backported as well.

**Comment by bendeutsch** [comment]
Would the database layer need to be compiled into Minetest? Or would it be possible to do this with C++-level plugins with `dlopen(3)` or stuff?

I haven't found any usage of `dlopen` in the C++ code, but there's the `Cpp API` – can that be used to dynamically load code?

**Comment by nerzhul** [comment]
i have a working auth db with multiple backends on my fork. I can backport it if needed. But if i remember it's already the case no ?

**Comment by nerzhul** [comment]
Hmm no, we did it for players, not auth. but it's easy to backport from my fork if needed, but i got some issues on my server these days i cannot dev

**Comment by bendeutsch** [comment]
Do you have a link to your port? I'd like to have a look at it.

**Comment by bendeutsch** [comment]
Pull request #7279 has just left "WIP" status and is an attempt to address this issue.

**Comment by sorcerykid** [comment]
> Not sure I still have the big auth file. And I'm terrified of upgrading minetest servers.

Hey @Megaf, I just published my Auth Redux mod (long overdue I know), which addresses all of the known issues. It's entirely Lua-based as I mentioned above. I also kept it backwards compatible, since there are still many 0.4.15 and 0.4.16 servers that might need a fix. You're welcome to look it over.

https://forum.minetest.net/viewtopic.php?f=9&t=20393&p=323768
